##逻辑回归主要用于这种二项分类问题，采用sigmoid函数作为预测函数，当x=0时，sigmoid函数的值为0.5，之后向两边趋近，因此它得到的结果都是非黑及白的。例如划分乳腺癌是为阴性还是阳性，就比较好划分

In [1]:
from sklearn.datasets import load_breast_cancer
cancer=load_breast_cancer()#加载乳腺癌数据
x=cancer.data#加载乳腺癌判别特征
y=cancer.target#两个特征，y=0时为阴性，y=1时为阳性
print('data shape:{0};no.positive:{1};no.negative:{2}'.format(x.shape,y[y==1].shape,y[y==0].shape))
print(cancer.data[0])#输出


data shape:(569, 30);no.positive:(357,);no.negative:(212,)
[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
 1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
 6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
 1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
 4.601e-01 1.189e-01]


##从中我们可以看出X中data一共有569个数据，30个特征。只是其关注特征其实为10个，即为
radius：半径，即病灶中心点离边界的平均距离

texture：纹理，灰度值的标准偏差

perimeter：周长，即病灶的大小

area：面积，反映病灶大小的一个指标

smoothness：平滑度，即半径的变化幅度

compactness：密实度，周长的平方除以面积的熵，再减一

concavity：凹度，凹陷部分轮廓的严重程度

concave points：凹点，凹陷轮廓的数量

symmetry：对称性

fractal dimension：分形维度
每一个有着对应的标准差与最大值，所以这里会有30个特征值。

In [4]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(x,y,test_size=0.2)#划分训练集和测试集
from sklearn.linear_model import LogisticRegression
model=LogisticRegression()#逻辑回归模型的建立
model=fit(X_train,y_train)#逻辑回归模型的训练
train_score=model.score(X_train,y_train)#训练的得分
test_score=model.score(X_test,y_test)#测试得分
print('train score:{train_score:.6f};test score:{test_score:.6f}'.format(train_score=train_score,test_score=test_score))

NameError: name 'fit' is not defined